[CFNetwork] Remove unneeded respondsToSelector checks

LumioseSil opened the pull request:
#### 8bb936f2feba3f9b47b4c2dcb55522da5dea179e
<pre>
[CFNetwork] Remove unneeded respondsToSelector checks
<a href="https://bugs.webkit.org/show_bug.cgi?id=279443">https://bugs.webkit.org/show_bug.cgi?id=279443</a>

Reviewed by Alexey Proskuryakov.

These respondToSelector checks all return true now on the supported OSes.

Remove CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY while we are at
it.

* Source/WTF/wtf/PlatformHave.h: Remove
  CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY.
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h: Ditto.
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
  (WebKit::NetworkSessionCocoa::NetworkSessionCocoa):
  (WebKit::NetworkSessionCocoa::addWebPageNetworkParameters):
  Remove unneeded respondToSelector checks.

Canonical link: <a href="https://commits.webkit.org/283625@main">https://commits.webkit.org/283625@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e19e657ce6c07b7c1a87ff57ec4b908684305fb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66798 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46173 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19420 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70833 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17931 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68916 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17706 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53509 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12067 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69865 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42488 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57792 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34143 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39160 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15184 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16285 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/59912 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61071 "Found 1 new API test failure: TestWebKitAPI.SleepDisabler.Unload (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15525 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72534 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66043 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10755 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14877 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60918 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10787 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57849 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61152 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14765 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8834 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2453 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/87810 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41980 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15445 "Found 4 new JSC stress test failures: jsc-layout-tests.yaml/js/script-tests/function-apply-aliased.js.layout-no-cjit, stress/json-stringify-inspector-check.js.no-llint, wasm.yaml/wasm/stress/cc-int-to-int-cross-module-with-exception.js.default-wasm, wasm.yaml/wasm/stress/cc-int-to-int-no-jit.js.default-wasm (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43057 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44240 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42800 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
```diff
@@ -1497,10 +1497,7 @@ - (void)URLSession:(NSURLSession *)session webSocketTask:(NSURLSessionWebSocketT
 
 #if HAVE(CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY)
-    // We still need to check the selector since CFNetwork updates and WebKit updates are separate
-    // on older macOS.
```
As you are touching this code anyway, please remove `CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY` everywhere.

aproskuryakov commented on Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
```diff
@@ -1950,8 +1945,7 @@ - (void)URLSession:(NSURLSession *)session webSocketTask:(NSURLSessionWebSocketT
     ASSERT(addResult2.isNewEntry);
     RetainPtr<NSURLSessionConfiguration> configuration = adoptNS([m_defaultSessionSet->sessionWithCredentialStorage.session.get().configuration copy]);
 #if HAVE(CFNETWORK_NSURLSESSION_ATTRIBUTED_BUNDLE_IDENTIFIER) && USE(APPLE_INTERNAL_SDK)
```
Sorry, I didn't notice at first that this was a different check. So we only need `USE(APPLE_INTERNAL_SDK)` here.

aproskuryakov commented on Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
```diff
@@ -1949,9 +1944,8 @@ - (void)URLSession:(NSURLSession *)session webSocketTask:(NSURLSessionWebSocketT
     auto addResult2 = m_perPageSessionSets.add(pageID, SessionSet::create());
     ASSERT(addResult2.isNewEntry);
     RetainPtr<NSURLSessionConfiguration> configuration = adoptNS([m_defaultSessionSet->sessionWithCredentialStorage.session.get().configuration copy]);
-#if HAVE(CFNETWORK_NSURLSESSION_ATTRIBUTED_BUNDLE_IDENTIFIER) && USE(APPLE_INTERNAL_SDK)
```
Do we still need `CFNETWORK_NSURLSESSION_ATTRIBUTED_BUNDLE_IDENTIFIER` in `PlatformHave.h`?

AZero13 replied:
> Do we still need `CFNETWORK_NSURLSESSION_ATTRIBUTED_BUNDLE_IDENTIFIER` in `PlatformHave.h`?

I removed it.

LumioseSil committed:
[CFNetwork] Remove unneeded respondsToSelector checks
https://bugs.webkit.org/show_bug.cgi?id=279443

Reviewed by Alexey Proskuryakov.

These respondToSelector checks all return true now on the supported OSes.

Remove CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY while we are at
it.

* Source/WTF/wtf/PlatformHave.h: Remove
  CFNETWORK_OVERRIDE_SESSION_COOKIE_ACCEPT_POLICY.
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h: Ditto.
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
  (WebKit::NetworkSessionCocoa::NetworkSessionCocoa):
  (WebKit::NetworkSessionCocoa::addWebPageNetworkParameters):
  Remove unneeded respondToSelector checks.

Canonical link: https://commits.webkit.org/283625@main
